### PR TITLE
Move poll to the backend.  Closes #491

### LINKED ISSUE #491: Move poll to Backend

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by mcovarr** [comment]
@scottfrazer and @kshakir

File: engine/src/test/scala/SimpleWorkflowActorSpec.scala [hunk 1]
```diff
@@ -25,7 +23,7 @@ class SimpleWorkflowActorSpec extends CromwellTestkitSpec {
     TestFSMRef(new WorkflowActor(descriptor, new LocalBackend(system)))
   }
 
-  val TestExecutionTimeout = 5.seconds.dilated
+  val TestExecutionTimeout = 10.seconds.dilated
```
Comment (Horneth) [inline engine/src/test/scala/SimpleWorkflowActorSpec.scala]: Hmm is it a "because I can" kind of change or is there a hidden reason ? :p

Comment (geoffjentry) [inline engine/src/test/scala/SimpleWorkflowActorSpec.scala]: This test has been failing for me locally for a while now for this exact reason

Comment (mcovarr) [inline engine/src/test/scala/SimpleWorkflowActorSpec.scala]: Yeah, I needed to change this as the test was taking ~7 seconds.

File: engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 2]
```diff
@@ -339,9 +340,11 @@ case class JesBackend(actorSystem: ActorSystem)
 
   override def stdoutStderr(backendCall: BackendCall): CallLogs = backendCall.stdoutStderr
 
+  /** WARNING returns a modified copy of the input jobDescriptor as part of the BackendCall result. */
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: This will be somewhat less disgusting once the `BackendCall` goes away

**Comment by scottfrazer** [comment]
:+1:

**Comment by kshakir** [comment]
:+1:

**Pull request merged**